src/main/java/org/spongepowered/api/service/economy/account/Account.java

```diff
@@ -372,8 +372,10 @@ default TransferResult transfer(Account to, Currency currency, BigDecimal amount
 
     /**
      * Deletes the account, removing all traces of the account having existed.
+     *
+     * @return Whether the deletion was successful.
      */
-    default void delete() {
-
+    default boolean delete() {
```

Comment:
This should indicate possible reasons for the failure of the deletion in the javadocs.

Reply:
"Implementation related issues". What else could even be a factor?

Reply:
Possible reasons include that the account doesn't exist in the first place, or that account deletion is not supported (default case).

I'd consider whether this boolean could be a new `AccountDeletionResult` (name pending) `CatalogType` that has the common reasons as failiure options, an `OTHER_FAILURE` (name also pending) option, as well as a success option, at least a plugin using this might have a better idea as to what happened.

@Flibio and @Erigitic might have an interest in this too as economy plugin authors.

Reply:
I suppose you mean on the method in the service primarily.

Reply:
I would remove this method and only keep the service one. It makes no sense to have a `delete` method on the `Account` object as you'd still have a reference to the account and so would have an object that is practically worthless to you. 

Better to just have it on the service where you don't have the object created at all.